### PR TITLE
Add attester/proposer slashing endpoints

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

Continuation to #809.

## Proposed Changes

Merges in latest master and updates the tests.

### CODE REVIEW

**Comment by paulhauner** [comment]
This has a conflict now sorry

**Comment by pawanjay176** [comment]
@paulhauner Fixed! :)

**Comment by paulhauner** [comment]
Thank you! Will jump onto this in a couple of hours :)

File: book/src/http_beacon.md [hunk 1]
```diff
@@ -7,106 +7,128 @@ beacon chain and also historical information about beacon blocks and states.
 
 HTTP Path | Description |
 | --- | -- |
+[`/beacon/attester_slashing`](#beaconattester_slashing) | Insert an attestesr slashing
```
Comment (paulhauner) [inline book/src/http_beacon.md]: Typo: attestesr

File: book/src/http_beacon.md [hunk 2]
```diff
@@ -7,106 +7,128 @@ beacon chain and also historical information about beacon blocks and states.
...
 [`/beacon/validators/active`](#beaconvalidatorsactive) | Get all active validators.
-[`/beacon/committees`](#beaconcommittees) | Get the shuffling for an epoch.
+[`/beacon/validators/all`](#beaconvalidatorsall) | Get all validators.
 
-## `/beacon/head`
+## `/beacon/attester_slashing`
 
-Requests information about the head of the beacon chain, from the node's
-perspective.
+Adds the specified `attester_slashing` to the operation pool.
```
Comment (paulhauner) [inline book/src/http_beacon.md]: Perhaps we could elaborate to this:

> Accepts an `attester_slashing` and verifies it. If it is valid, it is added to the operations pool for potential inclusion in a future block. Returns an 400 error if the `attester_slashing` is invalid.

File: book/src/http_beacon.md [hunk 3]
```diff
@@ -7,106 +7,128 @@ beacon chain and also historical information about beacon blocks and states.
...
 
 | Property | Specification |
 | --- |--- |
-Path | `/beacon/head`
-Method | GET
+Path | `/beacon/attester_slashing`
+Method | POST
 JSON Encoding | Object
 Query Parameters | None
 Typical Responses | 200
```
Comment (paulhauner) [inline book/src/http_beacon.md]: We could add 400 here

File: book/src/http_beacon.md [hunk 4]
```diff
@@ -188,9 +210,212 @@ Typical Responses | 200, 404
...
+    {
+        "beacon_block_root": "0x41ed5b253c4fc841cba8a6d44acbe101866bc674c3cfa3c4e9f7388f465aa15b",
+        "beacon_block_slot": 38375
+    }
+]
+```
+
+## `/beacon/proposer_slashing`
+
+Adds the specified `proposer_slashing` to the operation pool.
```
Comment (paulhauner) [inline book/src/http_beacon.md]: Perhaps we could elaborate again:


> Accepts a `proposer_slashing` and verifies it. If it is valid, it is added to the operations pool for potential inclusion in a future block. Returns an 400 error if the `proposer_slashing` is invalid.

File: book/src/http_beacon.md [hunk 5]
```diff
@@ -188,9 +210,212 @@ Typical Responses | 200, 404
...
+
+### HTTP Specification
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/proposer_slashing`
+Method | POST
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200
```
Comment (paulhauner) [inline book/src/http_beacon.md]: I think we could add 400 here too

**Comment by pawanjay176** [comment]
Fixed!

**Comment by paulhauner** [comment]
Thank you!!

**Pull request merged**